vars/capsuleSystemTests.groovy

```diff
@@ -0,0 +1,207 @@
+        directory: value.name,
+        credentialsId: config.gitCredentialsId,
+        timeout: 2,
+      ])
+    }]}
+  }
+
+  stage('build binaries') {
+    def binaries = [
+      [ repository: 'vegacapsule', name: 'vegacapsule', packages: './main.go' ],
+      [ repository: 'vega', name: 'vega', packages: './cmd/vega/' ],
+      [ repository: 'data-node', name: 'data-node', packages: './cmd/data-node/' ],
+      [ repository: 'vegawallet', name: 'vegawallet', packages: './main.go' ],
+    ]
+    
+    parallel binaries.collectEntries{value -> [value.name, { buildGoBinary(value.repository,  testDirectoryPath + '/' + value.name, value.packages) }]}
+  }
+  
+  stage('prepare network') {
+    dir('system-tests') {
+      sh 'cp -r ./vegacapsule/multisig-setup ' + testDirectoryPath
+      sh 'cp ./vegacapsule/capsule_config.hcl ' + testDirectoryPath + '/config_system_tests.hcl'
+    }
+
+    dir ('tests/multisig-setup') {
+      timeout(time: 5, unit: 'MINUTES') {
+        ansiColor('xterm') {
+          sh 'npm install'
+        }
+      }
+    }
+    
+    dir ('tests') {
+        sh 'daemonize -o ' + testDirectoryPath + '/nomad.log -c ' + testDirectoryPath + ' -p ' + testDirectoryPath + '/vegacapsule_nomad.pid ' + testDirectoryPath + '/vegacapsule nomad'
+    }
+
+    dir('system-tests/scripts') {
+      timeout(time: 5, unit: 'MINUTES') {
+        ansiColor('xterm') {
+          sh 'make check'
+          sh 'make prepare-test-docker-image'
+          sh 'make build-test-proto'
+        }
+      }
+    }
+  }
+
+  stage('start the network') {
+    dir('tests') {
+      try {
+        withCredentials([
+          usernamePassword(credentialsId: 'github-vega-ci-bot-artifacts', passwordVariable: 'TOKEN', usernameVariable:'USER')
+        ]) {
+          sh 'echo -n "' + TOKEN + '" | docker login https://ghcr.io -u "' + USER + '" --password-stdin'
+        }
+        timeout(time: 5, unit: 'MINUTES') {
+          ansiColor('xterm') {
+            sh './vegacapsule network bootstrap --config-path ./config_system_tests.hcl --home-path ' + testDirectoryPath + '/testnet'
+          }
+        }
+      } catch (e) {
+        throw e
+      } finally {
+        sh 'docker logout https://ghcr.io'
+      }
+      sh './vegacapsule nodes ls-validators --home-path ' + testDirectoryPath + '/testnet > ' + testDirectoryPath + '/testnet/validators.json'
+      sh 'mkdir -p ' + testDirectoryPath + '/testnet/smartcontracts'
+      sh './vegacapsule state get-smartcontracts-addresses --home-path ' + testDirectoryPath + '/testnet > ' + testDirectoryPath + '/testnet/smartcontracts/addresses.json'
+    }
+  }
+
+  stage('setup multisig contract') {
+    dir ('tests/multisig-setup') {
+      timeout(time: 2, unit: 'MINUTES') {
+        ansiColor('xterm') {
+          sh 'node main.js "' + testDirectoryPath + '/testnet/smartcontracts/addresses.json" "' + testDirectoryPath + '/testnet/validators.json"'
+        }
+      }
+    }
+  }
+
+  try {
```

Comment:
Maybe use declarative syntax for such complex operations? You will reduce code base drastically